content/articles/predicting-future-stock-prices-using-aima-model-in-r/index.md

```diff
@@ -0,0 +1,181 @@
+We need to install caTools library using; 
+
+`install.packages(“caTools”)` 
+
+Activate the library caTools by:
+
+`library(caTools)` 
+
+To select our train data, we use;
+
+`train_data<-AMAZON_diff[1:3355]`
+
+![differencing log data](/engineering-education/predicting-future-stock-prices-using-arima-model-in-r/difference.jpg)
+
+### Building ARIMA model 
+ARIMA model in R is found in the package ‘forecast’ which we will first install and then activate as follows;
+
+`install.packages(“forecast”)`
+
+`library(forecast)`
+
+Auto.arima is used to generate the ARIMA model
+
+
+![generating ARIMA model](/engineering-education/predicting-future-stock-prices-using-arima-model-in-r/auto.jpg)
+
+To check the summary for chosen best fit ARIMA model, we use;
+
+`summary(arima_mode)`
+
+![summary for our model](/engineering-education/predicting-future-stock-prices-using-arima-model-in-r/fit.jpg)
+
+We then check for any residual in our ARIMA model. and judging by the Ljung-Box test, we conclude that the p-value > 0.05 (insignificant) means that the model’s residuals are independent and not autocorrelated which means we don't have to do volatility modeling using models like Garch, commonly used on financial data with heteroscedasticity problem.
+![checking for residuals](/engineering-education/predicting-future-stock-prices-using-arima-model-in-r/hetero.jpg)
+
+Suggesting that ARIMA(0,0,2) is our model.
+
+The plot of our Residuals from our ARIMA model shows that our forecast for 100 days ahead shows a straight line indicating that our ARIMA model fits well as it is supposed to follow a normal distribution and should be stationary.  
+![residuals fom ARIMA](/engineering-education/predicting-future-stock-prices-using-arima-model-in-r/from.jpg)
+
+### Fitting ARIMA model and forecasting
+
+Now, to fit the model into the training data set, we use;
+
+`arima<-arima(train_data, order=c(0, 0, 2)`
+
+`summary(arima)`
+
+![fitting our model](/engineering-education/predicting-future-stock-prices-using-arima-model-in-r/model.jpg)
+
+
+Now, we can make our forecast of the next 100 days using `forecast` package with h=100
+
+And we can plot our forecast using 
+`plot(forecast)`
+
+![plotting our forecast](/engineering-education/predicting-future-stock-prices-using-arima-model-in-r/plotting.jpg)
+
+
+And then check residuals in our model using
+ `checkresiduals(arima)` 
+
+![checking residual](/engineering-education/predicting-future-stock-prices-using-arima-model-in-r/residual.jpg)
+
+![our arima](/engineering-education/predicting-future-stock-prices-using-arima-model-in-r/fore.jpg)
+
+Our forecast will be;
+
+![our forecast](/engineering-education/predicting-future-stock-prices-using-arima-model-in-r/forecast.jpg)
+
+This shows that in the next 100 days, there will be a rise in AMAZON’s stock prices with a slight downward movement in the next few days and then an almost steady rise.
+
+### Conclusion 

```

Comment:
In this section; Include any next or final steps summarizing the tasks the audience just completed with possible next steps.